runtime/shared/src/main/scala/org/http4s/googleapis/runtime/ComputeMetadata.scala

```diff
@@ -16,63 +16,213 @@
 
 package org.http4s
 package googleapis.runtime
-
+import cats.effect.Concurrent
 import cats.effect.Temporal
+import cats.effect.std.Env
+import cats.syntax.all._
+import com.comcast.ip4s.IpLiteralSyntax
+import fs2.io.IOException
+import io.circe.Decoder
 
+import circe.jsonOf
 import client.Client
 import auth.AccessToken
 import syntax.all._
+import org.http4s.googleapis.runtime.Options.Standard
+import org.http4s.googleapis.runtime.Options.Full
+import org.http4s.googleapis.runtime.Options.FullWithLicense
 
 trait ComputeMetadata[F[_]] {
+
+  /** The GCP project id.
+    * @see
+    *   https://cloud.google.com/compute/docs/metadata/predefined-metadata-keys
+    */
   def getProjectId: F[String]
+
+  /** The zone where this VM is located. This value has the following format:
+    * projects/PROJECT_NUM/zones/ZONE
+    *
+    * @see
+    *   https://cloud.google.com/compute/docs/metadata/predefined-metadata-keys#instance-attributes-metadata
+    */
   def getZone: F[String]
+
+  /** https://cloud.google.com/compute/docs/metadata/predefined-metadata-keys
+    *
+    * @see
+    *   The ID of the VM. This is a unique, numerical ID that is generated by Compute Engine.
+    *   This is useful for identifying VMs if you don't use VM names.
+    */
   def getInstanceId: F[String]
-  def getClusterName: F[String]
-  def getContainerName: F[String]
-  def getNamespaceId: F[String]
```

Comment:
cluster-name, container-name and namespace-id are not defined in the official document
https://cloud.google.com/compute/docs/metadata/predefined-metadata-keys#instance-attributes-metadata